.github/workflows/07-image-publish.yaml

```diff
@@ -0,0 +1,59 @@
+name: Build & Publish Docker Image to GHCR
+
+on:
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+defaults:
+  run:
+    working-directory: ./docker-app
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+
+    # This are permissions for the GITHUB_TOKEN
+    permissions:
+      contents: read
+      packages: write # Allows pushing to GHCR
+      attestations: write
+      id-token: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Log in to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+     
+      - name: Build and push Docker image
+        id: push
+        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
+        with:
+          context: ./docker-app
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+      
+      - name: Generate artifact attestation
```

Comment:
Your previous incarnation of this file where you called docker push directly would probably have worked if you have made the docker image name all lowwer case but you wouldn't have got the benefit of the attestation (increasing traceability and trust) for your image.